core/src/test/scala/com/raphtory/BaseRaphtoryAlgoTest.scala

```diff
@@ -1,61 +1,64 @@
 package com.raphtory
 
-import java.io.File
-import java.nio.charset.StandardCharsets
 import com.google.common.hash.Hashing
 import com.raphtory.algorithms.api.GraphAlgorithm
 import com.raphtory.algorithms.api.OutputFormat
 import com.raphtory.components.graphbuilder.GraphBuilder
 import com.raphtory.components.spout.Spout
-import com.raphtory.components.spout.SpoutExecutor
 import com.raphtory.config.PulsarController
 import com.raphtory.deploy.Raphtory
-import com.raphtory.client.RaphtoryGraph
-import com.raphtory.serialisers.PulsarKryoSerialiser
-import com.typesafe.config.Config
 import com.typesafe.config.Config
-import com.typesafe.config.ConfigFactory
-import com.typesafe.config.ConfigValueFactory
 import com.typesafe.scalalogging.Logger
+import org.apache.commons.io.FileUtils
 import org.apache.pulsar.client.api.Consumer
 import org.apache.pulsar.client.api.Message
-import org.apache.pulsar.client.api.Schema
-import org.scalactic.source
 import org.scalatest.BeforeAndAfter
 import org.scalatest.BeforeAndAfterAll
 import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
 import org.slf4j.LoggerFactory
 
+import java.io.File
+import java.nio.charset.StandardCharsets
 import scala.language.postfixOps
 import scala.reflect.ClassTag
-import scala.util.Random
 import scala.reflect.runtime.universe._
 
-abstract class BaseRaphtoryAlgoTest[T: ClassTag: TypeTag]
+abstract class BaseRaphtoryAlgoTest[T: ClassTag: TypeTag](deleteResultAfterFinish: Boolean = true)
         extends AnyFunSuite
-        with BeforeAndAfterAll {
-  val logger: Logger = Logger(LoggerFactory.getLogger(this.getClass))
-
-  setup()
+        with BeforeAndAfter
+        with BeforeAndAfterAll
+        with Matchers {
 
-  Thread.sleep(5000)
+  val outputDirectory: String = "/tmp/raphtoryTest"
+  val logger: Logger          = Logger(LoggerFactory.getLogger(this.getClass))
 
-  val spout        = setSpout()
-  val graphBuilder = setGraphBuilder()
+  val spout: Spout[T]               = setSpout()
+  val graphBuilder: GraphBuilder[T] = setGraphBuilder()
 
-  val graph            =
-    if (batchLoading) Raphtory.batchLoadGraph[T](spout, graphBuilder)
+  val graph =
+    if (batchLoading()) Raphtory.batchLoadGraph[T](spout, graphBuilder)
     else Raphtory.streamGraph[T](spout, graphBuilder)
-  val conf             = graph.getConfig()
-  val pulsarController = new PulsarController(conf)
 
-  val pulsarAddress: String =
-    conf.getString("raphtory.pulsar.broker.address") //conf.getString("Raphtory.pulsarAddress")
-  val deploymentID: String = conf.getString("raphtory.deploy.id")
+  val conf: Config          = graph.getConfig()
+  val pulsarAddress: String = conf.getString("raphtory.pulsar.broker.address")
+  val deploymentID: String  = conf.getString("raphtory.deploy.id")
+  val pulsarController      = new PulsarController(conf)
```

Comment:
If we are cleaning things, it may be good to wrap this stuff into a `beforeAll` so the graph is only initialised when tests start (right now all graphs are being initialised when tests are discovered).